src/systemctl/systemctl.c

```diff
@@ -4562,6 +4571,17 @@ static int show_one(
         if (r < 0)
                 return log_error_errno(r, "Failed to get properties: %s", bus_error_message(&error, r));
 
+        r = bus_message_map_all_properties(reply, property_map, &property_info);
+        if (r < 0)
+                return log_error_errno(r, "Failed to map properties: %s", bus_error_message(&error, r));
+
+        if (streq_ptr(property_info.load_state, "not-found") && streq_ptr(property_info.active_state, "inactive"))
+                return 4;
```

Comment:
You changed the 3 to 4, but you still didn't answer the question, why this cannot be -EXISTS or -ESRCH instead?